# Unique DUOX2+ACE2+ small cholangiocytes are pathogenic targets for primary biliary cholangitis


## System requirements

The source code was written manily by R and should be run on systems that support R. R version 3.6.3 was use to test the code. R packages denpendencies include: Seurat (version 3.1.2), Harmony (version 1.0), ClusterProfiler (version 4.4.4). CellPhoneDB (version 2.1.0) was used for the cell-cell interaction analysis.

## Installation guide

Please install the R software and R packages following the guide in their home page or github.

* R: https://www.r-project.org/
* Seurat: http://satijalab.org/seurat/
* Harmony: https://github.com/immunogenomics/harmony
* ClusterProfiler: https://bioconductor.org/packages/release/bioc/html/clusterProfiler.html
* CellPhoneDB: https://github.com/Teichlab/cellphonedb

## Demo

* Seurat is used for single-cell RNA sequencing data analysis. Mainly workflow contains quality control, sample aggregating, dimension reduction, clustering and visualization. The expected outputs include cell clusters, gene expression matrix and different plots for visualization. You can find the demo here: https://satijalab.org/seurat/vignettes.html.

* Harmony is mainly used to process batch effect. The result of harmony analysis will be used in the following dimension reduction. You can find the demo here: https://github.com/immunogenomics/harmony/blob/master/vignettes/.

* ClusterProfiler performs enrichment analysis based on GO or KEGG database. The output of this package include enriched functions and related statistics. You can follow the demo here: http://yulab-smu.top/clusterProfiler-book/.

* CellPhoneDB performs cellullar interaction analysis and the visualization of the result. You can get the strength of interactions and the related statistics. To understand the result you can refer to here: https://github.com/Teichlab/cellphonedb/blob/master/Docs/RESULTS-DOCUMENTATION.md. You can follow the demo here: https://github.com/Teichlab/cellphonedb.


## Instructions for use

### `main_analysis.R`

The following is the main script for data analysis including data integration, cell annotation, visialization, cell-cell interaction and enrichment analysis.

#### Data integration

##### Input

* `exp_list` The path of 10X expression matrix corresponding to the sample.

##### Output

* `umap.cluster.pdf` UMAP plot of the integrated data.

* `.diffgenes.xls` The differentially expressed genes list of each cluster.

* `integrated_cluster.rds` The Seurat object after integration.


#### Annotation

##### Input

* `celltypes` A vector storing cell type infomation.

##### Output

* `umap.label.pdf` Umap plot after adding cell type label.

* `integrated_add_celllabel.rds` The Seurat object after annotation and adding cell label.


#### Visializtion

##### Input

* `features` Genes need to be displayed.

##### Ouput

Figures displaying genes expression (violin plot, dot plot and heatmap).


#### Cell-cell interaction

##### Input

* `inputdata` The path of Seurat object rds used for cellphoneDB.

##### Output

The result was generated by cellphoneDB. To understand the result, you can refer to https://github.com/Teichlab/cellphonedb/blob/master/Docs/RESULTS-DOCUMENTATION.md.


#### Enrichment analysis

##### Input 

* `diffgene` Gene list used for enrichment analysis.

* `GO/KEGG_term` A vector storing description of terms you want to display.

##### Output

* `Enrichment_GO/KEGG.xls` The result of enrichment analysis.

* `Enrichment_GO/KEGG.pdf` The dot plot of the term you want to display in result.


### `sig_score.R`

This script was used to calculate Sig score and display the result.

#### Input

* `clusterrds` Integrated Seruat object rds before annotation.

* `marker` Marker gene list with the first colum is celltype and the second cloumn is marker genes corresponding to the celltype.

* `cluster` A table with the first column is celltype, and the second column is cluster number corresponding to the celltype.

#### Output

* `_sig_score_dotplot.pdf` The dot plot of the result.

* `_data.xls` Sig score result.
